Userland/Libraries/LibJS/Runtime/RegExpConstructor.h

```diff
@@ -17,13 +18,42 @@ class RegExpConstructor final : public NativeFunction {
     explicit RegExpConstructor(GlobalObject&);
     virtual void initialize(GlobalObject&) override;
     virtual ~RegExpConstructor() override = default;
+    void set_captured_values(ThrowCompletionOr<Array*> const& captured_values) { m_captured_values = captured_values; }
+    void set_input(ThrowCompletionOr<Value> const& input) { m_input = input; }
+    void set_last_match(ThrowCompletionOr<Value> const& last_match) { m_last_match = last_match; }
+    void set_last_paren(ThrowCompletionOr<Value> const& last_paren) { m_last_paren = last_paren; }
+    void set_right_context(ThrowCompletionOr<Value> const& right_context) { m_right_context = right_context; }
+    void set_left_context(ThrowCompletionOr<Value> const& left_context) { m_left_context = left_context; }
 
     virtual ThrowCompletionOr<Value> call() override;
-    virtual ThrowCompletionOr<Object*> construct(FunctionObject& new_target) override;
+    virtual ThrowCompletionOr<Object*> construct(FunctionObject&) override;
 
 private:
+    virtual void initialize_static_props(GlobalObject&);
     virtual bool has_constructor() const override { return true; }
-
+    ThrowCompletionOr<Value> get_captured_value(VM&, GlobalObject&, size_t);
+    ThrowCompletionOr<Array*> m_captured_values { nullptr };
+    ThrowCompletionOr<Value> m_input { js_string(vm(), "") };
+    ThrowCompletionOr<Value> m_last_match { js_string(vm(), "") };
+    ThrowCompletionOr<Value> m_last_paren { js_string(vm(), "") };
+    ThrowCompletionOr<Value> m_right_context { js_string(vm(), "") };
+    ThrowCompletionOr<Value> m_left_context { js_string(vm(), "") };
```

Comment:
No one is visiting these, so they'll be deleted if a GC collect event happens, also, I'm not sure why the these are stored as ThrowCompletionOr, the exception shouldn't be stored here.

Reply:
OK